op-node/rollup/derive/channel_bank.go

```diff
@@ -141,26 +143,52 @@ func (ib *ChannelBank) Read() (data []byte, err error) {
 	return data, nil
 }
 
-func (ib *ChannelBank) Step(ctx context.Context, outer Progress) error {
-	if changed, err := ib.progress.Update(outer); err != nil || changed {
-		return err
+// Step does the advancement for the channel bank.
+// Channel bank as the first non-pull stage does it's own progress maintentance.
+// When closed, it checks against the previous origin to determine if to open itself
+func (ib *ChannelBank) Step(ctx context.Context, _ Progress) error {
+	// Open ourselves
+	// This is ok to do b/c we would not have yielded control to the lower stages
+	// of the pipeline without being completely done reading from L1.
+	if ib.progress.Closed {
+		if ib.progress.Origin != ib.prev.Origin() {
+			ib.progress.Closed = false
+			ib.progress.Origin = ib.prev.Origin()
+			return nil
+		}
 	}
 
-	// If the bank is behind the channel reader, then we are replaying old data to prepare the bank.
-	// Read if we can, and drop if it gives anything
-	if ib.next.Progress().Origin.Number > ib.progress.Origin.Number {
-		_, err := ib.Read()
+	skipIngest := ib.next.Progress().Origin.Number > ib.progress.Origin.Number
+	outOfData := false
+
+	if data, err := ib.prev.NextData(ctx); err == io.EOF {
+		outOfData = true
+	} else if err != nil {
 		return err
+	} else {
+		ib.IngestData(data)
 	}
 
 	// otherwise, read the next channel data from the bank
 	data, err := ib.Read()
 	if err == io.EOF { // need new L1 data in the bank before we can read more channel data
-		return io.EOF
+		if outOfData {
+			if !ib.progress.Closed {
+				ib.progress.Closed = true
+				return nil
+			}
+			return io.EOF
+		} else {
+			return nil
+		}
 	} else if err != nil {
 		return err
+	} else {
+		if !skipIngest {
+			ib.next.WriteChannel(data)
+			return nil
+		}
 	}
-	ib.next.WriteChannel(data)
 	return nil
 }
```

Comment:
Ideally, we should strive to return something useful to the upper level.

Meaning that we should replay the whole amount of "old data" and then get some new one to return.

But: why was the decomposition in steps picked in the first place? Is there an idea that purposefully limiting / timeouting a step as a whole is useful?

One argument I could see is that while we may individually timeout every blocking call, a logic error could cause a lot (or even, an infinite amount) of them to be made, in which case we'd want to timeout the whole thing. But are timeouts really suppose to protect us against infinite loops of our own making? Probably not.

In general, I what I envision as the new system, a call to the engine queue would prepare a new PayloadAttributes, which could take significant time. Two issues:
- Timeouts as specified above.
     - We can't really do timeouts within the pipeline because such timeouts involve concurrency, which are not supported by Cannon! 
         - Timeouts in the fetcher operations are still fine. Should we just ensure the code is correct and go with this.
     - While we're pulling data and trying to create a payload attribute, we're insensitive to new information including re-orgs. This means we might not react to re-org as fast as possible.
         - We could make the fetcher interface return an error the first time it is called after a reorg is detected, in order to abort the pipeline early.